wgengine/monitor/monitor_windows.go

```diff
@@ -7,273 +7,159 @@ package monitor
 import (
 	"context"
 	"errors"
-	"fmt"
-	"runtime"
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
 	"sync"
-	"syscall"
-	"time"
-	"unsafe"
-
-	"golang.org/x/sys/windows"
-	"tailscale.com/net/interfaces"
 	"tailscale.com/types/logger"
-)
-
-const (
-	pollIntervalSlow = 15 * time.Second
-	pollIntervalFast = 3 * time.Second
-	pollFastFor      = 30 * time.Second
+	"time"
 )
 
 var (
-	iphlpapi                 = syscall.NewLazyDLL("iphlpapi.dll")
-	notifyAddrChangeProc     = iphlpapi.NewProc("NotifyAddrChange")
-	notifyRouteChangeProc    = iphlpapi.NewProc("NotifyRouteChange")
-	cancelIPChangeNotifyProc = iphlpapi.NewProc("CancelIPChangeNotify")
-)
-
-const (
-	_STATUS_PENDING = 0x00000103 // 259
-	_STATUS_WAIT_0  = 0
+	errClosed = errors.New("closed")
 )
 
 type unspecifiedMessage struct{}
 
 func (unspecifiedMessage) ignore() bool { return false }
 
-type pollStateChangedMessage struct{}
-
-func (pollStateChangedMessage) ignore() bool { return false }
-
 type messageOrError struct {
 	message
 	error
+	evt string
 }
 
 type winMon struct {
-	ctx             context.Context
-	cancel          context.CancelFunc
-	messagec        chan messageOrError
-	logf            logger.Logf
-	pollTicker      *time.Ticker
-	lastState       *interfaces.State
-	closeHandle     windows.Handle // signaled upon close
-	goroutineDoneCh chan struct{}  // closed when awaitIPAndRouteChanges goroutine has stopped
-
-	mu            sync.Mutex
-	lastNetChange time.Time
-	inFastPoll    bool // recent net change event made us go into fast polling mode (to detect proxy changes)
+	logf                  logger.Logf
+	ctx                   context.Context
```

Comment:
this is also guarded by mu it seems, so it'd seem like it'd go after the mu. (And restore the blank line before the mu line)

But I like the old style instead: one global ctx+cancel that doesn't mutate, and all sends+receives select on it too, then cancel it in Close.

Reply:
done